components/collator/src/provider.rs

```diff
@@ -633,7 +609,10 @@ impl CollationSpecialPrimariesValidated<'_> {
         // into Compiler Explorer shows that the panic
         // is optimized away.
         #[expect(clippy::indexing_slicing)]
-        let field = self.compressible_bytes[usize::from(b >> 4)];
+        let field = self
+            .last_primaries
+            .get(MaxVariable::Currency as usize + usize::from(b >> 4))
+            .unwrap_or_else(|| CollationSpecialPrimaries::HARDCODED_FALLBACK[usize::from(b >> 4)]);
```

Comment:
To keep the above comment true, shouldn't the compressibility bits be referenced as a fixed-length array in the collator constructor instead of doing slice access on every call to this method?